react/modules/SequentialTaskQueue.ts

```diff
@@ -153,78 +155,109 @@ export class SequentialTaskQueue {
    * @returns {CancellablePromiseLike<any>} A promise that can be used to await or cancel the task.
    */
-  push(task: Function, options?: TaskOptions): CancellablePromiseLike<any> {
-    if (this._isClosed)
+  public push(
+    task: TaskEntryFn,
+    options?: TaskOptions
+  ): CancellablePromiseLike<any> {
+    if (this._isClosed) {
       throw new Error(`${this.name} has been previously closed`)
+    }
+
     const taskEntry: TaskEntry = {
       callback: task,
-      args:
-        options && options.args
-          ? Array.isArray(options.args)
-            ? options.args.slice()
-            : [options.args]
-          : [],
+      args: options?.args
+        ? Array.isArray(options.args)
+          ? options.args.slice()
+          : [options.args]
+        : [],
       timeout:
-        options && options.timeout !== undefined
-          ? options.timeout
-          : this.defaultTimeout,
+        options?.timeout !== undefined ? options.timeout : this.defaultTimeout,
       cancellationToken: {
         cancel: (reason?) => this.cancelTask(taskEntry, reason),
       },
       resolve: undefined,
       reject: undefined,
     }
+
     taskEntry.args.push(taskEntry.cancellationToken)
     this.queue.push(taskEntry)
     this.scheduler.schedule(() => this.next())
     const result = (new Promise((resolve, reject) => {
       taskEntry.resolve = resolve
       taskEntry.reject = reject
     }) as any) as CancellablePromiseLike<any>
+
     result.cancel = (reason?: any) =>
       taskEntry.cancellationToken.cancel!(reason)
+
     return result
   }
 
   /**
    * Cancels the currently running task (if any), and clears the queue.
    * @returns {Promise} A Promise that is fulfilled when the queue is empty and the current task has been cancelled.
    */
-  cancel(): PromiseLike<any> {
-    if (this.currentTask)
+  public cancel(): PromiseLike<any> {
+    if (this.currentTask) {
       this.cancelTask(this.currentTask, cancellationTokenReasons.cancel)
+    }
+
     const queue = this.queue.splice(0)
+
     // Cancel all and emit a drained event if there were tasks waiting in the queue
     if (queue.length) {
-      queue.forEach(task =>
+      queue.forEach((task) =>
         this.cancelTask(task, cancellationTokenReasons.cancel)
       )
       this.emit(sequentialTaskQueueEvents.drained)
     }
+
     return this.wait()
   }
 
+  public indexOf(task: TaskEntryFn): number {
+    if (this.currentTask?.callback === task) {
+      return 0
+    }
+
+    const queueIndex = this.queue
+      // Skip cancelled tasks
+      .filter((taskEntry) => !taskEntry.cancellationToken?.cancelled)
+      .findIndex((taskEntry) => taskEntry.callback === task)
+
+    if (queueIndex < 0) {
+      return queueIndex
+    }
+
+    // Add 1 here because the queue is always missing the first
+    // element, which is stored in `this.currentTask`, so we need
+    // to shift the index by 1.
+    return queueIndex + 1
+  }
```

Comment:
Do you think it's worth to add an unit test for this?

Reply:
sure! I'll fix the broken unit tests as well